
```{r}
library(dplyr)
library(DESeq2) 
library(stringr)
library(ggplot2)
library(RColorBrewer)
library(ComplexHeatmap)
library(ggdendro)
library(circlize)
library(EnhancedVolcano)
library(ggvenn)
library(reshape2)
library(patchwork)
library(clusterProfiler)
library(org.Mm.eg.db)
library(enrichplot)

library(annotate)

```



```{r read raw count matrix}
Egr1_HIP <- read.csv("./c_union_rawCounts_newSamples.csv")

rownames(Egr1_HIP) <- Egr1_HIP$X
raw_counts.HIP <- Egr1_HIP[c(2:25)]
```



```{r build sample table for DESeq2}
sample_table.HIP <- data.frame(do.call(rbind, str_split(colnames(raw_counts.HIP),"_")))
colnames(sample_table.HIP) <- c("sample_id","Exp_condition","Sex")


sample_table.HIP$condition <- paste(sample_table.HIP$Exp_condition,sample_table.HIP$Sex,sep = "_")
sample_table.HIP$Exp_condition <- factor(sample_table.HIP$Exp_condition, levels = c("wtNeg","wtPos","ffNeg","ffPos"))

sample_table.HIP$Condition[sample_table.HIP$Exp_condition == "wtNeg"] <- "wt,Cre(-)"
sample_table.HIP$Condition[sample_table.HIP$Exp_condition == "wtPos"] <- "wt,Cre(+)"
sample_table.HIP$Condition[sample_table.HIP$Exp_condition == "ffNeg"] <- "Egr1,Cre(-)"
sample_table.HIP$Condition[sample_table.HIP$Exp_condition == "ffPos"] <- "Egr1,Cre(+)"

sample_table.HIP$Condition <- factor(sample_table.HIP$Condition, levels = c("wt,Cre(-)","wt,Cre(+)","Egr1,Cre(-)","Egr1,Cre(+)"))

colnames(raw_counts.HIP) <- substr(colnames(raw_counts.HIP), start = 1, stop = 6)

```

```{r adjusting the order for futher use}
sample_order <- c(sample_table.HIP$sample_id[c(sample_table.HIP$Sex == "F" & sample_table.HIP$Condition == "wt,Cre(-)")],
  sample_table.HIP$sample_id[c(sample_table.HIP$Sex == "F" & sample_table.HIP$Condition == "wt,Cre(+)")],
  sample_table.HIP$sample_id[c(sample_table.HIP$Sex == "F" & sample_table.HIP$Condition == "Egr1,Cre(-)")],
  sample_table.HIP$sample_id[c(sample_table.HIP$Sex == "F" & sample_table.HIP$Condition == "Egr1,Cre(+)")],
  sample_table.HIP$sample_id[c(sample_table.HIP$Sex == "M" & sample_table.HIP$Condition == "wt,Cre(-)")],
  sample_table.HIP$sample_id[c(sample_table.HIP$Sex == "M" & sample_table.HIP$Condition == "wt,Cre(+)")],
  sample_table.HIP$sample_id[c(sample_table.HIP$Sex == "M" & sample_table.HIP$Condition == "Egr1,Cre(-)")],
  sample_table.HIP$sample_id[c(sample_table.HIP$Sex == "M" & sample_table.HIP$Condition == "Egr1,Cre(+)")])

rownames(sample_table.HIP) <- sample_table.HIP$sample_id

sample_table.HIP <- sample_table.HIP[sample_order,]
raw_counts.HIP <- raw_counts.HIP[sample_table.HIP$sample_id]
```




```{r annoattion file}

# The annotation is generated by applying the following code to the genome gtf file
# cat mouse.gtf | awk 'BEGIN{FS="\t"}{split($9,a,";"); if($3~"gene") print a[1]"\t"a[3]"\t"$1":"$4"-"$5"\t"a[5]"\t"$7}' | sed 's/gene_id "//' | sed 's/gene_id "//' | sed 's/gene_biotype "//'| sed 's/gene_name "//' | sed 's/gene_biotype "//' | sed 's/"//g' | sed 's/ //g' | sed '1igene_id\tGeneSymbol\tChromosome\tClass\tStrand' > mouse_gene_annotation_table_old.txt

trans <- read.delim("./Mus_musculus.GRCm39.107_gene_annotation_table.txt")
rownames(trans) <- trans$gene_id

trans <- trans[!(trans$GeneSymbol %in% c("gene_sourceensembl","gene_sourcehavana","gene_sourcehavana_tagene","gene_sourceensembl_havana")),]

```


```{r}
raw_counts.HIP$gene_symbol <- trans[rownames(raw_counts.HIP),]$GeneSymbol
raw_counts.HIP <- aggregate(raw_counts.HIP[-25], by = list(GENESYMBOL = raw_counts.HIP$gene_symbol), FUN = sum)
rownames(raw_counts.HIP) <- raw_counts.HIP$GENESYMBOL
raw_counts.HIP <- raw_counts.HIP[2:25]

raw_counts.HIP <- raw_counts.HIP[rowSums(raw_counts.HIP)>100,]
length(rownames(raw_counts.HIP)[rowSums(raw_counts.HIP)<100])
```


```{r}
dds <- DESeqDataSetFromMatrix(countData = raw_counts.HIP,
                              colData =  sample_table.HIP, 
                              design = ~condition)
dds <- DESeq(dds)
```

